argilla/tests/unit/test_settings/test_settings_templates.py

```diff
@@ -0,0 +1,35 @@
+import pytest
+from unittest import mock
+from argilla.settings._templates import DefaultSettingsMixin
+
+
+class TestDefaultSettingsMixin:
+    def test_for_document_classification(self):
```

Comment:
maybe the phrasing `document_classification` is a bit weird? Also, I would also consider adding image cat and token cat as basic defaults.

Reply:
Yeah, I can see that the naming is odd. We should iterate on those

My aim is to make a few generic enough methods that could be updated into variations, and avoid an exhaustive list of ML tasks. I.e. `document_classification` could have have alternative fields for image or chat. 

In a subsequent PR, I would like to add these:
- `mapping` in settings. This would make the default templates flexible to column naming. 
- `Features` >> `Settings` : we select and update a 'Settings` class based on the values and names of `datasets.Features`

With these features, the settings templates could be limited because we can get information from the dataset.

Reply:
@burtenshaw , perhaps content classification would be better?

Reply:
I agree that we would need a mapping to truly avoid the difficulties because otherwise, any mis-named columns/field/question should still be re-defined, right?

Reply:
@burtenshaw, I agree the features to setting would be very interesting.

Reply:
> @burtenshaw , perhaps `content_classification` or simply `classification` 

I think these overlap with `token_classification`

> would be better? Similarly, should we rename to `ranking` / `rating`?

This makes sense. What about `for_ranking`,`for_rating`, `for_labels`, `for_spans` ?